src/Util/Config.php

```diff
@@ -48,10 +57,23 @@ class Config
     public function __construct(array $options = [])
     {
         $this->validateKeys($options);
-        $this->setStringValue(self::API_URL, $options);
-        $this->setStringValue(self::SDK_IDENTIFIER, $options);
-        $this->setStringValue(self::SDK_VERSION, $options);
-        $this->setHttpClient($options);
+
+        foreach ($options as $key => $value) {
```

Comment:
Now iterating of the provided config so that each setter method no longer needs to check whether the a property is set